test/jdk/java/awt/Choice/SelectCurrentItemTest/SelectCurrentItemTest.java

```diff
@@ -20,115 +20,139 @@
 import java.awt.Choice;
-import java.awt.Robot;
+import java.awt.EventQueue;
 import java.awt.Frame;
-import java.awt.BorderLayout;
-import java.awt.AWTException;
-import java.awt.Point;
-import java.awt.Dimension;
+import java.awt.Rectangle;
+import java.awt.Robot;
 import java.awt.event.InputEvent;
-import java.awt.event.ItemListener;
-import java.awt.event.WindowListener;
 import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.WindowAdapter;
 import java.awt.event.WindowEvent;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/*
+ * @test
+ * @bug 4902933 8197810
+ * @summary Test that selecting the current item does not send an ItemEvent
+ * @key headful
+ * @run main SelectCurrentItemTest
+ */
+public final class SelectCurrentItemTest
+        extends WindowAdapter
+        implements ItemListener {
+    private static Frame frame;
+    private static Choice choice;
 
-public class SelectCurrentItemTest implements ItemListener, WindowListener {
-    //Declare things used in the test, like buttons and labels here
-    private Frame frame;
-    private Choice theChoice;
-    private Robot robot;
+    private final Robot robot;
 
-    private CountDownLatch latch = new CountDownLatch(1);
-    private volatile boolean passed = true;
+    private final CountDownLatch windowOpened = new CountDownLatch(1);
+    private final CountDownLatch mouseClicked = new CountDownLatch(1);
 
-    private void init()
-    {
-        try {
-            robot = new Robot();
-            robot.setAutoDelay(500);
-        } catch (AWTException e) {
-            throw new RuntimeException("Unable to create Robot. Test fails.");
-        }
+    private volatile boolean itemStateChanged;
+
+    private SelectCurrentItemTest() throws AWTException {
+        robot = new Robot();
+        robot.setAutoDelay(500);
+    }
 
+    private void createUI() {
         frame = new Frame("SelectCurrentItemTest");
         frame.setLayout(new BorderLayout());
-        theChoice = new Choice();
+
+        choice = new Choice();
         for (int i = 0; i < 10; i++) {
-            theChoice.add(new String("Choice Item " + i));
+            choice.add("Choice Item " + i);
         }
-        theChoice.addItemListener(this);
-        frame.add(theChoice);
+        choice.addItemListener(this);
+        choice.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                System.out.println("mouseClicked()");
+                mouseClicked.countDown();
+            }
+        });
+
+        frame.add(choice, BorderLayout.CENTER);
+
         frame.addWindowListener(this);
 
-        frame.setLocation(1,20);
-        robot.mouseMove(10, 30);
+        frame.setLocationRelativeTo(null);
         frame.pack();
```

Comment:
When frame.setSize() is used, the mouse pointer on the choice drop-down button is no longer a resize indicator and the click event is received. But I guess you are planning to leave this as-s now so as to not mask the external issue.

Reply:
> When frame.setSize() is used, the mouse pointer on the choice drop-down button is no longer a resize indicator and the click event is received.

I don't understand how using `frame.setSize` resolve the problem? Both `pack` and `setSize` call `reshape` to apply the new location and size.

Did you mean `frame.setResizable(false)`?

> But I guess you are planning to leave this as-is for now to test when the external fix is available for [JDK-8322653](https://bugs.openjdk.org/browse/JDK-8322653)

We can modify this test to avoid JDK-8322653.

I added a panel to the frame to move the mouse cursor away from the edge of the frame while I verified the test would pass.

Reply:
> I don't understand how using frame.setSize resolve the problem? Both pack and setSize call reshape to apply the new location and size.

When the frame width and height is increased then the mouse is no longer a resize indicator which makes the 1st click (on the choice dropdown button) work but the 2nd click on the choice item needs to be adjusted accordingly.

I believe frame.setResizable(false) or adding a panel as you suggested is a better and easier option.

> We can modify this test to avoid JDK-8322653

Since there are already few tests problemlisted to track the external issue JDK-8322653, I think any of the above workarounds that you suggested can be added to avoid JDK-8322653.